Steady-state solver

nils-schween opened the pull request:
The steady-state solver is now in a ready-to-merge state. I finished the
corresponding example and I improved a little bit. 

An important point is that I do change the interface of sapphire, because I
introduce a new flag time_evolution. This required a change of all examples. I
did change them.

Let me know what you think.

By the way this addresses issue #7

floschulze reviewed (approved):
Thanks for implementing this feature @nils-schween!

I have pushed a few commits with minor style/typo fixes. Feel free to either squash them (`git rebase -i --autosquash main` or leave them like is.

One small thing you might want to change before merging is the short description of the new example in the `examples.md`. So far it only copies the text of the parallel-shock example.

An other question is, if it is worth implementing a test-case for the steady-state solver.

A last comment to the example: I like it very much. You have also done some improvements compared to the parallel-shock example as well. We should maybe think about implementing these general improvements in the 'normal' parallel-shock example, so the new example can focus more on the steady-state solver.
(But its probably better to wait till the scaling-factor is merged as well, and then go through all examples again to bring them up to date)

nils-schween commented:
Thank you, Florian. I rebased and squashed all the fixup commits together. I reverted one commit. I prefer the displayed equations in the table. And yes, a test would be great. But I suggest, we leave it as it is for now and we discuss next week when the scaled distribution is also available.
